packages/core/src/derived.ts

```diff
@@ -86,29 +86,61 @@ export class Derived<T> {
   compute(): void {
     const prevContext = getCurrentContext();
     const prevComputation = getRunningComputation();
-    const context = setupCurrentContext(this);
+
+    const context: Array<ReactiveValue> = [];
+    setCurrentContext(context);
     setRunningComputation(this);
 
+    unlinkObservers(this);
+
     const result = this.computeFn();
 
     this.sources = context;
 
+    for (let i = 0; i < this.sources.length; i++) {
+      const source = this.sources[i];
+      if (source) {
+        if (source.observers) {
+          source.observers.push(this);
+        } else {
+          source.observers = [this];
+        }
+      }
+    }
+
     setCurrentContext(prevContext);
     setRunningComputation(prevComputation);
 
     if (result !== this.lastValue) {
-      markUpdate(this, DIRTY);
+      markUpdates(this, DIRTY);
       this.lastValue = result;
     }
   }
+
+  markUpdate(status: NOTCLEAN) {
+    // CLEAN < STALE < DIRTY
+    if (this.status < status) {
+      this.status = status;
+
+      if (this.observers) {
+        for (let i = 0; i < this.observers.length; i++) {
+          this.observers[i]?.markUpdate(STALE);
+        }
+      }
+    }
+  }
 }
 
 export function createDerived<T>(fn: () => T, label?: string): Derived<T> {
   return new Derived(fn, label);
 }
 
-function createLogger(label?: string): (...data: Array<unknown>) => void {
-  return (...data: Array<unknown>) => {
-    console.log(label ? `[${label}]` : '', ...data);
-  };
+// function createLogger(label?: string): (...data: Array<unknown>) => void {
+//   return (...data: Array<unknown>) => {
+//     console.log(label ? `[${label}]` : '', ...data);
+//   };
+// }
+
+export function isDerived(v: unknown): v is Derived<unknown> {
+  return v instanceof Derived;
```

Comment:
Another tiny perf improvement: use the tag I added in #21 to do this without the `instanceof` check. It won't add up to much, I expect, but it'll be *some* small win!